src/flyte/_resources.py

```diff
@@ -249,7 +254,7 @@ def __post_init__(self):
             raise ValueError("GPU quantity must be at least 1")
 
 
-def GPU(device: GPUType, quantity: GPUQuantity, partition: A100Parts | A100_80GBParts | None = None) -> Device:
+def GPU(device: GPUType, quantity: GPUQuantity, partition: A100Parts | A100_80GBParts | H200Parts | None = None) -> Device:
```

Comment:
Maybe worth creating a type alias for these MIG types.

Reply:
Thinking we can group this union types into a single alias. As we add more partition types, function def will look cleaner. Def not critical.